src/main/java/com/codegym/jrugotom5/service/AdvertService.java

```diff
@@ -30,4 +35,21 @@ public List<AdvertDTO> getAdvertsByDateRange(LocalDate from, LocalDate to) {
                 .map(advert -> modelMapper.map(advert, AdvertDTO.class))
                 .collect(Collectors.toList());
     }
+
+    public AdvertDTO update(AdvertDTO advertDTO) {
```

Comment:
> What will be if your DTO will have ID with null value? Add check for this purpose in very begining. In the same IF block you can add check if entity exists with such ID in DB (so if there is no advert you will throw Your Exception)

there is @NotNull for id in dto

Reply:
> there is NotNull annotation for id in dto

Yeap) Validation didn't work, so i got the error during test :D i agree with you.
But still we need to check if the Entity exists in the DB. Otherwise, we will get an error during the findById method.